pkg/router/seller.go

```diff
@@ -8,262 +8,780 @@ import (
-		return c.NoContent(http.StatusOK)
 	}
 }
 
-// @Summary Seller edit shop info
-// @Description Edit shop name, description, visibility.
-// @Tags Seller, Shop
-// @Param  image_id       query     string  false  "name search by q"  Format(email)
-// @Param  name           query     string  false  "name search by q"  Format(email)
-// @Param  Description    query     string  false  "name search by q"  Format(email)
-// @Param  enable         query     string  false  "name search by q"  Format(email)
-// @Produce json
-// @Success 200
-// @Failure 401
-// @Router /seller [patch]
+// @Summary		Seller edit shop info
+// @Description	Edit shop name, description, visibility.
+// @Tags			Seller, Shop
+// @Param			image_id	body	string	true	"update image UUID"
+// @Param			name		body	string	true	"update shop name"	minlength(6)
+// @Param			description	body	string	true	"update description"
+// @Param			enabled		body	bool	true	"update enabled status"
+// @Produce		json
+// @success		200	{object}	db.SellerUpdateInfoRow
+// @Failure		400	{object}	echo.HTTPError
+// @Failure		500	{object}	echo.HTTPError
+// @Router			/seller/info [patch]
 func sellerEditInfo(pg *db.DB, logger *zap.SugaredLogger) echo.HandlerFunc {
 	return func(c echo.Context) error {
-
-		return c.NoContent(http.StatusOK)
+		var username string = "user1"
+
+		var param db.SellerUpdateInfoParams
+		if err := c.Bind(&param); err != nil {
+			logger.Error(err)
+			return echo.NewHTTPError(http.StatusBadRequest)
+		}
+		param.SellerName = username
+		shopInfo, err := pg.Queries.SellerUpdateInfo(c.Request().Context(), param)
+		if err != nil {
+			logger.Error(err)
+			return echo.NewHTTPError(http.StatusInternalServerError)
+		}
+		return c.JSON(http.StatusOK, shopInfo)
 	}
 }
 
-// @Summary Seller get available tag
-// @Description Get all available tags for shop.
-// @Tags Seller, Shop, Tag
-// @Produce json
-// @Success 200
-// @Failure 401
-// @Router /seller/tag [get]
+// @Summary		Seller get available tag
+// @Description	Get all available tags for shop.
+// @Tags			Seller, Shop, Tag
+// @Param			name	body	string	true	"search tagname start with"	minlength(1)
+// @Produce		json
+// @success		200	{array}		db.SellerSearchTagRow
+// @Failure		400	{object}	echo.HTTPError
+// @Failure		500	{object}	echo.HTTPError
+// @Router			/seller/tag [get]
 func sellerGetTag(pg *db.DB, logger *zap.SugaredLogger) echo.HandlerFunc {
 
 	return func(c echo.Context) error {
-
-		return c.NoContent(http.StatusOK)
+		var username string = "user1"
+		var tagPerPage int32 = 20
+
+		var param db.SellerSearchTagParams
+		if err := c.Bind(&param); err != nil {
+			logger.Error(err)
+			return echo.NewHTTPError(http.StatusBadRequest)
+
+		}
+		if param.Name == "" || hasRegexSpecialChars(param.Name) {
+			return echo.NewHTTPError(http.StatusBadRequest)
+		}
+		param.SellerName = username
+		param.Name = "^" + param.Name
+		param.Limit = tagPerPage
+		tags, err := pg.Queries.SellerSearchTag(c.Request().Context(), param)
```

Comment:
constantly polling the DB when typing isn't optimal performance wise, it would be better if this was done in-memory or something like Redis